src/versionbits.cpp

```diff
@@ -9,9 +9,11 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
 {
     int nPeriod = Period(params);
     int nThreshold = Threshold(params);
-    int min_activation_height = MinActivationHeight(params);
     int64_t nTimeStart = BeginTime(params);
     int64_t nTimeTimeout = EndTime(params);
+    int64_t nHeightStart = BeginHeight(params);
```

Comment:
We partially implement BIP8 here as just BIP9 with 2 differences: 
1. Start & Timeout is set by block height instead of actual time
2. If signaling threshold is not met by timeout height, we force activation of the feature during the next window instead of failing activation

Major differences from BIP8 spec:
1. We do not use `minimum_activation_height`
2. `lockinontimeout` is assumed true and not currently configurable